apps/admin-gui/src/app/shared/components/dialogs/add-role-dialog/add-role-dialog.component.ts

```diff
@@ -74,6 +88,22 @@ export class AddRoleDialogComponent implements OnInit {
   }
 
   addRole(): void {
-    this.submitForm.emit({ role: this.selectedRule, entity: this.selectedEntity });
+    if (this.selectedRule.primaryObject === 'Facility') {
+      this.submitForm.emit({
+        role: this.selectedRule,
+        entities: this.selectedFacilities.selected.map((ef) => ef.facility),
+      });
+    } else {
+      this.submitForm.emit({ role: this.selectedRule, entities: this.selected.selected });
+    }
+  }
+
+  resetSelection(selectedRule: RoleManagementRules): void {
+    this.selectedRule = selectedRule;
+    this.selected.clear();
+    this.selectedFacilities.clear();
+    this.filterValue = '';
+    this.loadObjects();
+    this.filterComponent.formControl = new UntypedFormControl('');
```

Comment:
1) There is an error when you try to set the value to a non-existing element (the assignment is called before the element is created) -> so you want to change the value just for the existing element.
![image](https://user-images.githubusercontent.com/70371638/210751649-038854d9-e112-4025-b1ab-1fc5921134fe.png)
2) You don't have to replace form control by new form control. You can just change the value like this:
```suggestion
    if (this.filterComponent) {
      this.filterComponent.formControl.setValue('');
    }
```